index.js

```diff
@@ -208,6 +208,9 @@ class Hypercore extends EventEmitter {
       // in practice, open an issue and we'll try to make a solution for it.
       throw SESSION_CLOSED('Cannot make sessions on a closing core')
     }
+    if (opts.checkout && !opts.name) {
```

Comment:
checkout !== undefined

Reply:
Fixed (now does explicit undefined checks for the others too)